core/src/main/java/org/eclipse/hono/util/ResourceIdentifier.java

```diff
@@ -188,7 +227,7 @@ public String getEndpoint() {
      */
     public String getTenantId() {
         if (resourcePath.length > IDX_TENANT_ID) {
-            return resourcePath[IDX_TENANT_ID];
+            return resourcePath[IDX_TENANT_ID].length() == 0 ? null : resourcePath[IDX_TENANT_ID];
```

Comment:
I would rather set the corresponding segments to `null` when creating the instance. In the end, this is supposed to an immutable ...